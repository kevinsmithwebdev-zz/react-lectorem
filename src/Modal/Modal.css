
.Modal {
  background: white;
  border: 1px solid #ccc;
  border-radius: 5px;
  box-shadow: -5px 5px 5px rgba(0, 0, 0, 0.2);
  left: 50%;
  opacity: 1;
  position: fixed;
  top: 50%;
  transform: scale(1);
  transform: translate(-50%, -50%);
  visibility: visible;
  min-width: 200px;
  max-width: 500px;
  z-index: 100;
  overflow: auto;
}

.Modal.off {
  opacity: 0;
  visibility: hidden;
  transform: scale(0.33);
  box-shadow: 5px 0 0 rgba(0, 0, 0, 0.2);
}

.Modal .header {
  background-color: #ddd;
  padding: 2px;
  margin: 0;
  display: flex;
  justify-content: space-between;
}

.Modal button {
  background-color: rgba(0, 0, 0, 0);
  border-width: 0;
  color: red;
  font-size: 12px;
  padding: 5px 10px;
}

.Modal .line {
  border-top: 1px solid #ddd;
  padding: 5px;
  text-align: left;
}

.Modal .line .label {
  font-weight: bold;
}
