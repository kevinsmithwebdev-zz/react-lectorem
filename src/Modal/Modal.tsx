import React from 'react';
import classes from './Modal.module.css';
import { LectInterface } from '../interfaces/index';
import { convertToTrainCase } from '../utils/case';
interface ModalData extends LectInterface {
  originalSentence: string,
  sentenceTranslation?: string,
}

export interface ModalInterface {
  data: ModalData | null,
  hideModal: Function,
}

export const renderLine = (label: string, thisData: string | undefined): JSX.Element  => {
  if (!thisData) {
    // @ts-ignore
    return null;
  }

  return (
    <div className='line' data-testid={ convertToTrainCase(label) }>
      <span className='label'>{`${label}: `}</span>
      <span className='data'>{thisData}</span>
    </div>
  );
};

const Modal: React.FC<ModalInterface> = ({ data, hideModal }) => {
  if (!data) {
    return null;
  }

  const handleClick = /* istanbul ignore next */  (_e: React.MouseEvent) => hideModal();

  return (
    <div data-testid='hint_modal' className={ classes.Modal }>
      <div className={ classes.header }>
        <h3>Helpful Hint</h3>
        <button data-testid='modal-close-button' className={ classes['toggle-button'] } onClick={ handleClick } type='button' aria-label='button'>
          <span role='img'>
            &#x274c;
          </span>
        </button>
      </div>

      { renderLine('Text', data.text) }
      { renderLine('Translation', data.translation) }
      { renderLine('Full Text', data.originalSentence) }
      { renderLine('Full Translation', data.sentenceTranslation) }
      { renderLine('Explanation', data.explanation) }
    </div>
  );
};

export default Modal;
