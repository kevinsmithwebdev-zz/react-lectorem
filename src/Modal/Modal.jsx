import React from 'react';
import './Modal.css';

export const renderLine = (label, thisData) =>
  !!thisData && (
    <div className='line'>
      <span className='label'>{`${label}: `}</span>
      <span className='data'>{thisData}</span>
    </div>
  );

const Modal = ({ data, hideModal }) => {
  if (!data) {
    return null;
  }
  return (
    <div className='Modal'>
      <div className='header'>
        <h3>Helpful Hint</h3>
        <button className='toggle-button' onClick={ hideModal } type='button'>
          <span role='img' aria-label='close'>
            &#x274c;
          </span>
        </button>
      </div>

      {renderLine('Text', data.text)}
      {renderLine('Translation', data.translation)}
      {renderLine('Full Text', data.originalSentence)}
      {renderLine('Full Translation', data.sentenceTranslation)}
      {renderLine('Explanation', data.explanation)}
    </div>
  );
};

export default Modal;
