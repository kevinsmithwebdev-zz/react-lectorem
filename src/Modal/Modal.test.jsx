import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Modal, { renderLine } from './Modal';

const mockLect = {
  start: 1.23,
  end: 4.56,
  text: 'Hola.',
  translation: 'Hello.',
  explanation: 'This is how we say hello,',
};
describe('Modal', () => {
  describe('the main component render', () => {
    const renderer = new ShallowRenderer();
    describe('empty', () => {
      it('should render without crashing', () => {
        renderer.render(<Modal />);
        expect(renderer.getRenderOutput()).toBeNull();
      });
    });

    describe('with data', () => {
      const props = {
        data: mockLect,
        hideModal: 'hideModal123',
      };

      it('should render without crashing', () => {
        renderer.render(<Modal { ...props } />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      });
    });
  });

  describe('renderLine', () => {
    describe('empty', () => {
      it('should render and not crash', () => {
        expect(renderLine()).toBe(false);
      });
    });
  });

  describe('renderLine', () => {
    describe('empty', () => {
      const label = 'label123';
      const thisData = 'thisData123';
      it('should render and not crash', () => {
        expect(renderLine(label, thisData)).toMatchSnapshot();
      });
    });
  });
});
