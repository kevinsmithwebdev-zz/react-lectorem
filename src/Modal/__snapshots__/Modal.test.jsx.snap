// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renderLine empty should render and not crash 1`] = `
<div
  className="line"
>
  <span
    className="label"
  >
    label123: 
  </span>
  <span
    className="data"
  >
    thisData123
  </span>
</div>
`;

exports[`Modal the main component render with data should render without crashing 1`] = `
<div
  className="Modal"
>
  <div
    className="header"
  >
    <h3>
      Helpful Hint
    </h3>
    <button
      className="toggle-button"
      onClick="hideModal123"
      type="button"
    >
      <span
        aria-label="close"
        role="img"
      >
        ‚ùå
      </span>
    </button>
  </div>
  <div
    className="line"
  >
    <span
      className="label"
    >
      Original Text: 
    </span>
    <span
      className="data"
    >
      Hola.
    </span>
  </div>
  <div
    className="line"
  >
    <span
      className="label"
    >
      Translation: 
    </span>
    <span
      className="data"
    >
      Hello.
    </span>
  </div>
  <div
    className="line"
  >
    <span
      className="label"
    >
      Explanation: 
    </span>
    <span
      className="data"
    >
      This is how we say hello,
    </span>
  </div>
</div>
`;
